docs/decouple-application-sync-user-using-impersonation

```diff
@@ -0,0 +1,269 @@
+---
+title: Decouple Control plane and Application Sync priveleges
+authors:
+  - "@anandf" # Authors' github accounts here.
+sponsors:
+  - Red Hat   # List all interested parties here.
+reviewers:
+  - "@alexmt"
+  - "@jannfis"
+approvers:
+  - "@alexmt"
+  - "@jannfis"
+
+creation-date: 2023-06-23
+last-updated: 2023-06-23
+---
+
+# Decouple Application Sync using Impersonation
+
+Application syncs in Argo CD have the same privileges as the Argo CD control plane. As a consequence, the Argo CD control plane privileges needs to match the tenant that needs the highest privileges. As an example, if an Argo CD instance has 10 Applications and only one of them needs admin privileges, the Argo CD control plane then must have admin privileges in order to successfully sync for that one Application. Argo CD provides a multi-tenancy model to limit what each Application can do even though the control plane has high privileges however that creates a large attack surface since if Argo CD is compromised, attackers would have cluster-admin access to the cluster.
+
+The goal of this proposal is to perform the Application sync as a different user using impersonation and use the service account provided in the cluster config purely for control plane tasks.
+
+
+
+## Open Questions [optional]
+
+- Should the restrictions imposed as part of the AppProjects be honored if the impersonation feature is enabled ?
+- Can an Application refer to a service account with elevated privileges like say  `cluster-admin`, `admin`, and service accounts used for running the ArgoCD controllers itself ?
+
+
+## Summary
+
+In a multi team/multi tenant environment, an application team is typically granted access to a namespace to self-manage their Applications in a declarative way. Current implementation of ArgoCD requires the ArgoCD Administrator to create an AppProject with access settings configured to replicate the RBAC resources that are configured for each team. This approach requires duplication of effort and also requires syncing the access between both to maintain the security posture. It would be desirable for users to use the existing RBAC rules without having to revert to Argo CD API to create and manage these Applications. One namespace per team, or even one namespace per application is what we are looking to address as part of this proposal.
+
+## Motivation
+
+This proposal would allow ArgoCD administrators to manage the cluster permissions using kubernetes native RBAC implementation rather than using complex configurations in `AppProjects` to restrict access to individual applications. By decoupling the priveleges required for application sync from the privelege required for ArgoCD control plane, the security requirement of providing least privelage can be achieved there by improving the security posture of ArgoCD. For implementing multi team/tenant use cases, this decoupling would be greatly beneficial.
```

Comment:
```suggestion
This proposal would allow Argo CD administrators to manage the cluster permissions using Kubernetes native RBAC implementation rather than using complex configurations in `AppProjects` to restrict access to individual applications. By decoupling the privileges required for application sync from the privileges required for the Argo CD control plane, the security requirement of providing least privilege can be achieved there by improving the security posture of ArgoCD. For implementing multi team/tenant use cases, this decoupling would be greatly beneficial.
```